### PR TITLE
StarRocks SQL Parser 修复和完善

### PR DESCRIPTION
**Pull request by guangchaozhou**
1.完善解析 Crete Table 语法，目前已经基本支持 StarRocks v3.1 版本及之前的建表语句的 SQL 解析。
2.对已有 StarRocks SQL Parser 的错误进行修复

### CODE REVIEW

**Comment by wenshao** [comment]
本地运行bvt/sql的testcase，修复不能通过的问题。构建日志中看到有如下错误：
```
2024-07-10T03:06:54.3928310Z [ERROR] Failures: 
2024-07-10T03:06:54.3929381Z [ERROR]   BigQueryResourceTest.bigquery_parse:14->SQLResourceTest.fileTest:96->SQLResourceTest.fileTest:127 expected:<[ASSERT (
2024-07-10T03:06:54.3929828Z 	SELECT COUNT(*) > 5
2024-07-10T03:06:54.3930041Z 	FROM UNNEST([1, 2, 3, 4, 5, 6]))
2024-07-10T03:06:54.3930497Z AS '''Table must contain more than 5 rows.'''];> but was:<[];>
2024-07-10T03:06:54.3930686Z [ERROR] Errors: 
2024-07-10T03:06:54.3931759Z [ERROR]   Issue5412.test_analyze:64 » UnsupportedOperation com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGAnalyzeStatement
2024-07-10T03:06:54.3932987Z [ERROR]   Issue5412.test_analyze_emptytable:39 » UnsupportedOperation com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGAnalyzeStatement
2024-07-10T03:06:54.3934208Z [ERROR]   Issue5412.test_vacuum:112 » UnsupportedOperation com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGVacuumStatement
2024-07-10T03:06:54.3935379Z [ERROR]   Issue5412.test_vacuum_empty:87 » UnsupportedOperation com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGVacuumStatement
2024-07-10T03:06:54.3936905Z [ERROR]   SparkResourceTest.test:19->SQLResourceTest.fileParse:71 » UnsupportedOperation com.alibaba.druid.sql.dialect.spark.ast.stmt.SparkCacheTableStatement
2024-07-10T03:06:54.3937867Z [ERROR]   Issue5905.test_parse_create:207 » Parser not supported.pos 192, line 6, column 16, token IDENTIFIER ENGINE
2024-07-10T03:06:54.3938049Z [INFO] 
2024-07-10T03:06:54.3938383Z [ERROR] Tests run: 6481, Failures: 1, Errors: 6, Skipped: 0
```

File: core/src/main/java/com/alibaba/druid/sql/SQLUtils.java [hunk 1]
```diff
@@ -81,6 +79,8 @@
 import java.util.function.Predicate;
 
 public class SQLUtils {
+    public static final Charset UTF8 = Charset.forName("UTF-8");
```
Comment (wenshao) [inline core/src/main/java/com/alibaba/druid/sql/SQLUtils.java:82]: 使用 java.nio.charset.StandardCharsets.UTF_8，不要使用Charset.forName

**Comment by wenshao** [comment]
不要修改和PR标题无关的内容

**Comment by guangchaozhou** [comment]
> 本地运行bvt/sql的testcase，修复不能通过的问题。构建日志中看到有如下错误：
> 
> ```
> 2024-07-10T03:06:54.3928310Z [ERROR] Failures: 
> 2024-07-10T03:06:54.3929381Z [ERROR]   BigQueryResourceTest.bigquery_parse:14->SQLResourceTest.fileTest:96->SQLResourceTest.fileTest:127 expected:<[ASSERT (
> 2024-07-10T03:06:54.3929828Z 	SELECT COUNT(*) > 5
> 2024-07-10T03:06:54.3930041Z 	FROM UNNEST([1, 2, 3, 4, 5, 6]))
> 2024-07-10T03:06:54.3930497Z AS '''Table must contain more than 5 rows.'''];> but was:<[];>
> 2024-07-10T03:06:54.3930686Z [ERROR] Errors: 
> 2024-07-10T03:06:54.3931759Z [ERROR]   Issue5412.test_analyze:64 » UnsupportedOperation com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGAnalyzeStatement
> 2024-07-10T03:06:54.3932987Z [ERROR]   Issue5412.test_analyze_emptytable:39 » UnsupportedOperation com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGAnalyzeStatement
> 2024-07-10T03:06:54.3934208Z [ERROR]   Issue5412.test_vacuum:112 » UnsupportedOperation com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGVacuumStatement
> 2024-07-10T03:06:54.3935379Z [ERROR]   Issue5412.test_vacuum_empty:87 » UnsupportedOperation com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGVacuumStatement
> 2024-07-10T03:06:54.3936905Z [ERROR]   SparkResourceTest.test:19->SQLResourceTest.fileParse:71 » UnsupportedOperation com.alibaba.druid.sql.dialect.spark.ast.stmt.SparkCacheTableStatement
> 2024-07-10T03:06:54.3937867Z [ERROR]   Issue5905.test_parse_create:207 » Parser not supported.pos 192, line 6, column 16, token IDENTIFIER ENGINE
> 2024-07-10T03:06:54.3938049Z [INFO] 
> 2024-07-10T03:06:54.3938383Z [ERROR] Tests run: 6481, Failures: 1, Errors: 6, Skipped: 0
> ```



> 本地运行bvt/sql的testcase，修复不能通过的问题。构建日志中看到有如下错误：
> 
> ```
> 2024-07-10T03:06:54.3928310Z [ERROR] Failures: 
> 2024-07-10T03:06:54.3929381Z [ERROR]   BigQueryResourceTest.bigquery_parse:14->SQLResourceTest.fileTest:96->SQLResourceTest.fileTest:127 expected:<[ASSERT (
> 2024-07-10T03:06:54.3929828Z 	SELECT COUNT(*) > 5
> 2024-07-10T03:06:54.3930041Z 	FROM UNNEST([1, 2, 3, 4, 5, 6]))
> 2024-07-10T03:06:54.3930497Z AS '''Table must contain more than 5 rows.'''];> but was:<[];>
> 2024-07-10T03:06:54.3930686Z [ERROR] Errors: 
> 2024-07-10T03:06:54.3931759Z [ERROR]   Issue5412.test_analyze:64 » UnsupportedOperation com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGAnalyzeStatement
> 2024-07-10T03:06:54.3932987Z [ERROR]   Issue5412.test_analyze_emptytable:39 » UnsupportedOperation com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGAnalyzeStatement
> 2024-07-10T03:06:54.3934208Z [ERROR]   Issue5412.test_vacuum:112 » UnsupportedOperation com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGVacuumStatement
> 2024-07-10T03:06:54.3935379Z [ERROR]   Issue5412.test_vacuum_empty:87 » UnsupportedOperation com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGVacuumStatement
> 2024-07-10T03:06:54.3936905Z [ERROR]   SparkResourceTest.test:19->SQLResourceTest.fileParse:71 » UnsupportedOperation com.alibaba.druid.sql.dialect.spark.ast.stmt.SparkCacheTableStatement
> 2024-07-10T03:06:54.3937867Z [ERROR]   Issue5905.test_parse_create:207 » Parser not supported.pos 192, line 6, column 16, token IDENTIFIER ENGINE
> 2024-07-10T03:06:54.3938049Z [INFO] 
> 2024-07-10T03:06:54.3938383Z [ERROR] Tests run: 6481, Failures: 1, Errors: 6, Skipped: 0
> ```

已修复

**Comment by wenshao** [comment]
1. 不要修改.gitignore文件
2. 测试的SQL是否能放到资源文件中，类似这个 https://github.com/alibaba/druid/blob/master/core/src/test/java/com/alibaba/druid/bvt/sql/spark/SparkResourceTest.java

**Comment by wenshao** [comment]
sqlserverstatementparser.java 这个类你没有修改，把格式化的内容回复回去吧

**Pull request merged**